### ISSUE TITLE
Background images not working or buggy.

### ISSUE THREAD

**Issue by iahim** [gnome-terminator/terminator#635]
**Describe the bug**
I have installed Terminator on a clean Linux Mint (mate and cinnamon) and Xubuntu. On all of them the bugs are consistent.
The background image feature works only for the default profile. If I add a new profile, and for the new profile i try to have another background image, that does not work. 
On the default profile - that displays the background image - there is still a bug: the image is being stretched with the window. For example my background image is  full hd, but my terminator window is only 200px wide. The entire image is squeezed inside the 200px.... I think the correct setting would be that the terminal should display only a part of the image (like it used to be before).... 
Now the image is distorted when i resize the terminator window.

**To Reproduce**
Add a new profile with a background image - that doesn't work.  I can see in the config file that the path of the image is correct,but the image is not rendered when I switch profiles.

**Expected behavior**
Each profile should have its own background image.

**Screenshots**

**Desktop (please complete the following information):**
Linux mint 21 (both mate and cinnamon) latest as of today.
Xubuntu latest - as of today.
Terminator version is 2.1.1

**Comment by MisterHarrow** [comment]
I have the same issue, the only thing the background sets is transparency level.

**Comment by iahim** [comment]
> 

Correct, for any other profile, instead of getting the BG image, you get only the % of transparency.

**Comment by MisterHarrow** [comment]
The background can be set on the default profile but not on any other profiles. Whether this is intended or not I don't know.

**Comment by iahim** [comment]
> The background can be set on the default profile but not on any other profiles. Whether this is intended or not I don't know.

Correct, only the default profile can have a BG.  For the moment I will switch to Konsole as a terminal emulator, the bg picture in Konsole works as expected for any profile.

**Comment by Vulcalien** [comment]
I think the issue about the background image not updating was fixed in this PR #551 (specifically this commit a06e7951f66175336872794fdb4f33a6ff804fb6). I would need to double check if it actually fixes that, but I'm pretty sure it does.

The way the image is drawn (stretch, scale keeping ratio...) is on the checklist in PR #612.

***

> The background can be set on the default profile but not on any other profiles.

If I remember correctly, the background is set only for the *first* loaded profile, whichever it is. But yeah, hopefully #551 fixed it.

**Comment by MisterHarrow** [comment]
I'm on version 2.1.1.

> If I remember correctly, the background is set only for the first loaded profile, whichever it is. But yeah, hopefully https://github.com/gnome-terminator/terminator/pull/551 fixed it.

The background image can be set on all profiles but only functions/loads on the default profile.

I realized yesterday that there's a couple of MS delay on my input on the default terminal when a background is activated as well. Not sure how to replicate that to show you though?

**Comment by Vulcalien** [comment]
> I realized yesterday that there's a couple of MS delay on my input

Indeed, the code that draws the image was very inefficient, but that was fixed in the same PR (#551).
This problem and the image not updating should both be fixed once there is a release.

Note to maintainers: keeping this issue open as a enhancement (image being stretched instead of keeping ratio), because (I think) all the bugs were fixed in #551
